### ISSUE TITLE
Owntone.Service failed due to 'Failed to start DAAP/DACP'

### ISSUE THREAD

**Issue by bitevader** [owntone/owntone-server#1753]
Please try to provide the following:

- steps to reproduce and/or logging of the error
- version of owntone
- platform

Steps to reproduce will greatly improve the chance of getting it fixed. If it is not possible then set the log level to debug (in /etc/owntone.conf) and try to get some logging of when the error happens. Don’t cut and paste lengthy log outtakes here on github. Instead, attach the log file.

**Comment by bitevader** [comment]
Hello
My target device is a Raspi 3 B. What I primarily want to do, is getting analog ins from turntable or cassette player to reach wireless Airplay-compatible speakers.
My first owntone installation followed standard procedure for bullseye 28.9.129
When it failed on DAAP I tried another installation following owntone build instructions / Quick version for Debian/Ubuntu. The failure turned out exactly as before.
The irony is that I had it working one or two weeks earlier, albeit without getting the fifo-pipe to work. I am at the end of my wisdom and mental resources now and more confused than ever.
2 more things that could have an influence:
1. I have darkice and icecast installed also, with which I got playing as mp3 on the computer, but not further. 
2. What role could be played by remnants of an attempted airtunes installation which ended with a build error? It isn't a prerequisit for owntone, right?
I am greatful for any help.

**Comment by bitevader** [comment]
No, no, its not completed!!!

**Comment by ejurgensen** [comment]
Please check the logs and post the relevant error messages

**Comment by bitevader** [comment]
Hi, nice to meet you!
Here's the log:
    main: OwnTone version 28.9 taking off
[2024-04-16 15:28:51] [  LOG]     main: Built with:
[2024-04-16 15:28:51] [  LOG]     main: - ffmpeg
[2024-04-16 15:28:51] [  LOG]     main: - Spotify
[2024-04-16 15:28:51] [  LOG]     main: - librespot-c
[2024-04-16 15:28:51] [  LOG]     main: - LastFM
[2024-04-16 15:28:51] [  LOG]     main: - Chromecast
[2024-04-16 15:28:51] [  LOG]     main: - MPD
[2024-04-16 15:28:51] [  LOG]     main: - Websockets
[2024-04-16 15:28:51] [  LOG]     main: - ALSA
[2024-04-16 15:28:51] [  LOG]     main: - Pulseaudio
[2024-04-16 15:28:51] [  LOG]     main: - Webinterface
[2024-04-16 15:28:51] [  LOG]     main: - Regex
[2024-04-16 15:28:51] [  LOG]     main: mDNS init
[2024-04-16 15:28:51] [  LOG]     mdns: Avahi state change: Client running
[2024-04-16 15:28:51] [  LOG]       db: Configured to use database file '/var/cache/owntone/songs3.db'
[2024-04-16 15:28:51] [  LOG]       db: Could not prepare statement: no such table: admin
[2024-04-16 15:28:51] [  LOG]       db: Could not prepare statement: no such table: admin
[2024-04-16 15:28:51] [  LOG]       db: Could not check database version, trying DB init
[2024-04-16 15:28:52] [  LOG]       db: Database OK with 0 active files and 6 active playlists
[2024-04-16 15:28:52] [  LOG]   laudio: Pulseaudio failed with error: Connection refused
[2024-04-16 15:28:52] [  LOG]   laudio: Error initializing Pulseaudio: Connection refused
[2024-04-16 15:28:52] [  LOG]     scan: Skipping library directory /srv/music, could not dereference: No such file or directory
[2024-04-16 15:28:52] [  LOG]     scan: Bulk library scan completed in 0 sec
[2024-04-16 15:28:52] [  LOG]      lib: Library init scan completed in 0 sec (0 changes)
[2024-04-16 15:58:21] [  LOG]     main: Got SIGTERM or SIGINT
[2024-04-16 15:58:21] [  LOG]     main: Stopping gracefully
[2024-04-16 15:58:21] [  LOG]     main: mDNS deinit
[2024-04-16 15:58:21] [  LOG]     main: Remote pairing deinit
[2024-04-16 15:58:21] [  LOG]     main: MPD deinit
[2024-04-16 15:58:21] [  LOG]     main: HTTPd deinit
[2024-04-16 15:58:21] [  LOG]     main: Player deinit
[2024-04-16 15:58:21] [  LOG]     main: Library scanner deinit
[2024-04-16 15:58:21] [  LOG]     main: Cache deinit
[2024-04-16 15:58:21] [  LOG]     main: Worker deinit
[2024-04-16 15:58:21] [  LOG]     main: Database deinit
[2024-04-16 15:58:21] [  LOG]     main: Exiting.
 and here's owntone status:
     Loaded: loaded (/lib/systemd/system/owntone.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Tue 2024-04-16 15:59:18 CEST; 5min ago
       Docs: man:owntone(8)
    Process: 3020 ExecStart=/usr/sbin/owntone -f (code=exited, status=1/FAILURE)
   Main PID: 3020 (code=exited, status=1/FAILURE)
        CPU: 574ms

Apr 16 15:59:18 raspberrypi systemd[1]: owntone.service: Scheduled restart job, restart counter is at 10.
Apr 16 15:59:18 raspberrypi systemd[1]: Stopped DAAP/DACP (iTunes), RSP and MPD server, supports AirPlay and Rem>
Apr 16 15:59:18 raspberrypi systemd[1]: owntone.service: Start request repeated too quickly.
Apr 16 15:59:18 raspberrypi systemd[1]: owntone.service: Failed with result 'exit-code'.
Apr 16 15:59:18 raspberrypi systemd[1]: Failed to start DAAP/DACP (iTunes), RSP and MPD server, supports AirPlay>

hope you can make something out of

**Comment by hacketiwack** [comment]
> [2024-04-16 15:28:52] [ LOG] scan: Skipping library directory /srv/music, could not dereference: No such file or directory

It seems that you didn't configure the directory containing your music.

How do you start OwnTone?

**Comment by bitevader** [comment]
I've started owntone with:
pi@raspberrypi:~ $ sudo service owntone restart

These are the settings in owntone.conf I've changed:

	uid = "root"
	# Database location
#	db_path = "/var/cache/owntone/songs3.db"

	# Database backup location
	# Uncomment and specify a full path to enable abilty to use REST endpoint
	# to initiate backup of songs3.db
#	db_backup_path = "/var/cache/owntone/songs3.bak"

#	admin_password = ""
	# Directories to index
	directories = { "/home/pi/Music" }

# Local audio output
audio {
	# Name - used in the speaker list in Remote
	nickname = "Raspi"

# Pipe output
# Allows OwnTone to output audio data to a named pipe
#fifo {
	nickname = "fifo"
	path = "/home/pi/Music/AUX"
#}

**Comment by bitevader** [comment]
Bold text was all commented with #.
I don't know why it showed like this after copying!

**Comment by hacketiwack** [comment]
> Bold text was all commented with #.
> I don't know why it showed like this after copying!

It's because the bug tracker of GitHub uses markdown in the editor. If you want to put the configuration file you have to surround it with (3) ``` (before and after your text on one line.

Regarding your logs, it seems that in between you have changed the location of your files. So the logs don't seem to be the correct ones.
Or, you're configuring OwnTone in the wrong configuration file. Just a wild guess.

**Comment by hacketiwack** [comment]
Now that I look at your configuration file, it seems to be quite invalid.
For example, the `audio` section is not terminated with a `}`.
Moreover, you commented out the `fifo` section, but only the beginning and the end. You have to comment out each lines.

```conf
#fifo {
#	nickname = "fifo"
#	path = "/home/pi/Music/AUX"
#}
```

**Comment by ejurgensen** [comment]
Yes, agree with @hacketiwack. The beginning of the config file actually tells you what you need to change. I suggest you leave the rest as it is.

I'm not sure how OwnTone even starts when the config file is invalid. I suspect you may have multiple installs and multiple config files?

However, the reason OwnTone stops is this one:

> [2024-04-16 15:58:21] [ LOG] main: Got SIGTERM or SIGINT

This means that some external process stopped OwnTone. I can't say why.

**Comment by bitevader** [comment]
My music files, if any, are in /home/pi/Music. What is "Database location" for?
The part of most importance for me is the fifo section, I guess. There I forgot to  comment out everything. What does SIGTERM or SIGINT mean? How can I find out what stopped it? And how about the "Failed to start DAAP/DACP" ?

**Comment by ejurgensen** [comment]
The config for fifo is for output, as it says. For input you don't need config. Just mkfifo somewhere in your library.

**Comment by bitevader** [comment]
Bingo! At last and at least owntone runs now and I can access the web interface. Now the fifo part is the next hurdle. I shall see ..
Thanks for the moment

**Comment by bitevader** [comment]
Hello again,
I am still unsure about the correspondence between the fifo I make using mkfifo -m666 /home/pi/Music/mystream.m3u, other .mp3 files in this directories and the owntone.conf file.
Should "# Directories to index" where I have music files and "# Database location" be the same?
I get only an empty queue named mystream and no mp3 files showing in the IF.
No sound at all so far!

Appendix: owntone.conf
[owntone.txt](https://github.com/owntone/owntone-server/files/15001290/owntone.txt)

**Comment by ejurgensen** [comment]
From the config file:

# For regular use, the most important setting to configure is "directories",
# which should be the location of your media. Whatever user you have set as
# "uid" must have read access to this location. If the location is a network
# mount, please see the README.

You really just need to do as the config file says. Don't change database location, let OwnTone use the default path.

I suggest you first focus on just getting a regular mp3 to play. Then afterwards you can look into the fifo stuff (btw, not sure why you added m3u to the fifo - it's not a m3u, so I suggest you don't do that).

**Comment by bitevader** [comment]
You can look at it as a great breakthrough that the stream is playing now. 
Owntone is the only approach from those I tried to reach this stage without having to install alot more software. Congratulations and thanks again for the help.
But then (aspirations rise with availability)
 - max volume quite low
 - STOP button holds only for short time then plays again
 - when controlled from iPhone: how sync volume slider with iPhone volume control button?
 - mp3 files still don't show in IF
Any suggestions?

**Comment by ejurgensen** [comment]
> max volume quite low

This is outside OwnTone's control

> STOP button holds only for short time then plays again

OwnTone automatically starts playback of fifo's when data is written to them. If you don't want that then disable pipe_autostart in the config (but then you manually need to start playback, of course).

> when controlled from iPhone: how sync volume slider with iPhone volume control button?

You can't, the UI is just a web interface, I don't think a web interface can get phone volume control events

> mp3 files still don't show in IF

Don't know

**Comment by hacketiwack** [comment]
> You can't, the UI is just a web interface, I don't think a web interface can get phone volume control events

Correct. To my knowledge, this is not possible to get volume control events from the device in HTML (see [here](https://stackoverflow.com/questions/41205465/how-to-detect-volume-up-button-click-event-using-javascript)).

**Comment by bitevader** [comment]
Hello, 
I am in danger of going nuts again. After having the pipe work, and Spotify activated, both appeared on the Web-IF.
All of a sudden after having tried to index a subdirectory to the "Music" directory inorder to maybe making visible .mp3 files, only Spotify was left at the IF. During this I've got an unexpected install-event (ending with a restart of the Raspberry Pi) of which I append a pullout of the dpkg.log. I don't know if the case has something to do with it. Anyhow I then reset owntone.conf to the last working state which is identical to the one I have added to this topic already except for the fifo-nickname where I removed the .m3u extension. When I open a stream to my "Music" directory, I still get the confirmation: 
"Recording raw data 'stdin' : Signed 16 bit Little Endian, Rate 44100 Hz, Stereo"
But where as before, the fifo file "mystream" was recognized as a Fifo type, it is now either of unknown, text or microsoft prompt-type. Mostly it is .txt which within seconds changes to unknown.
Maybe the problem has something to do with Spotify even after logging out? 
[Unknown Pkg Install Event.txt](https://github.com/owntone/owntone-server/files/15054545/Unknown.Pkg.Install.Event.txt)
